apps/microsoft-teams/frontend/src/utils/msGraphApi.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  AccountInfo,
+  IPublicClientApplication,
+  InteractionRequiredAuthError,
+} from '@azure/msal-browser';
+import { graphConfig } from '@configs/authConfig';
+
+export interface MsGraphApiErrorType {
+  message: string;
+  status: number;
+}
+
+export class MsGraphApiError extends Error {
+  status: number;
+
+  constructor(res: MsGraphApiErrorType) {
+    super(res.message ?? '');
+    this.status = res.status;
+  }
+}
+
+interface OrganizationalBrandingSuccessResponse {
+  '@odata.context': string;
+  cdnList: string[];
+  faviconRelativeUrl: string;
+  squareLogoRelativeUrl: string;
+}
+
+/**
+ * This class is used to interact with the Microsoft Graph API.
+ * Specifically we use this API to get the organization name and logo to display after a user authenticates
+ * @param instance IPublicClientApplication, comes from useMsal hook
+ * @param account AccountInfo, comes from useMsal hook
+ */
+class MsGraph {
+  readonly baseUrl: string;
+  readonly instance: IPublicClientApplication;
+  readonly account: AccountInfo;
+  readonly tenantId: string;
+
+  constructor(instance: IPublicClientApplication, account: AccountInfo) {
+    this.baseUrl = graphConfig.graphOrgEndpoint;
+    this.instance = instance;
+    this.account = account;
+    this.tenantId = account.tenantId;
+  }
+
+  /**
```

Comment:
love to see the jsdocs!

Reply:
Yes, it was a great suggestion from you for us to use this more!